slacgismo/docs-browser
# Add Github Auth

This PR address issue #3 

# Current issues
1. Github auth

# Code changes
1. Add navbar, error modal, and login form components to index.html
2. Modify main.js to handle login events, use basic authentication to communicate with the Github API, and display a modal popup on errors.

# Documentation changes
N/A

# Validation suggestions
1. Validate that login works by logging into your account.
2. Validate that login persists on page refreshes.  It is currently not set to persist across browser close/open, but that can be changed quite easily by switching all instances of sessionStorage to localStorage
3. Ensure Authorization header is being set (after logging in) on Github API (api.github.*) requests by looking at the Chrome devTools networking tab.
4. Ensure Authorization header is *NOT* being set for AJAX requests that are not to the Github API (eg, raw.githubusercontent.*).
5. Validate error modal works by repeatedly hitting the Github API while you are not logged in.  After enough requests, an error modal should display when there is a 403 error.
